[Response Ops][Alerting] Alert deletion

jcger opened the pull request:
> [!WARNING]
> This PR is no longer valid. We are closing it due to requirements changes

## Feature Flag

```yml
# config/kibana.dev.yml
xpack.trigger_actions_ui.enableExperimental: ['alertDeletionSettingsEnabled']
```

## Summary

This PR is comprised of multiple sub-PRs:

**Rules Settings**
- Form UI - https://github.com/elastic/kibana/pull/209993
- Rules setting - https://github.com/elastic/kibana/pull/211046
- API to create/update rules settings - https://github.com/elastic/kibana/pull/211488 
- Preview API + UI - https://github.com/elastic/kibana/pull/211952 (Depends on Task definition PR)

**Alert Deletion Task**
- Task definition - https://github.com/elastic/kibana/pull/211987

jcger committed:
[Response Ops][Alerting] Alert deletion - New rule setting form (#209993)

## Summary

Closes https://github.com/elastic/kibana/issues/209257

To gain access to this feature, add
`xpack.trigger_actions_ui.enableExperimental:
['alertDeletionSettingsEnabled']` feature flag to your
`config/kibana.dev.yml` file

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jcger committed:
Merge branch 'main' into alerting/alerts-deletion

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into alerting/alerts-deletion

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into alerting/alerts-deletion

ymao1 committed:
[Response Ops][Alerting] Adding alert deletion rules setting (#211046)

Note that this merges into a feature branch!

Towards https://github.com/elastic/kibana/issues/208615

## Summary

Adds alerts deletion rule settings. This is added as a subfeature
privilege under the existing Rules Settings feature privilege and is
available in all environments.

This PR includes the schema changes required for the rules settings
saved object along with client code for getting and updating the fields.

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into alerting/alerts-deletion

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/278234)
* Commit: c3c8b8900df4bbb6a060da5aab3997f6506e7892
* [Cloud Deployment](https://kibana-pr-211384.kb.us-west2.gcp.elastic-cloud.com)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/278234","number":278234,"commit":"c3c8b8900df4bbb6a060da5aab3997f6506e7892"}],"number":278234}
buildkite-pr-comment-->

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into alerting/alerts-deletion

ymao1 committed:
Merge branch 'alerting/alerts-deletion' of github.com:elastic/kibana into alerting/alerts-deletion

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/280358)
* Commit: 77f20c9413f897deb865c318c20d1e1d1b08649a
* [Cloud Deployment](https://kibana-pr-211384.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #278234](https://buildkite.com/elastic/kibana-pull-request/builds/278234) succeeded c3c8b8900df4bbb6a060da5aab3997f6506e7892

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280358","number":280358,"commit":"77f20c9413f897deb865c318c20d1e1d1b08649a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/278234","number":278234,"commit":"c3c8b8900df4bbb6a060da5aab3997f6506e7892"}],"number":280358}
buildkite-pr-comment-->

ymao1 committed:
Merging in main

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283225)
* Commit: c37488b6c6f446f2fabefe921200e20e2b76e120

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/283225#01958a50-e019-4d51-92fc-0ddccad5e9a0)

### History
* :yellow_heart: [Build #280358](https://buildkite.com/elastic/kibana-pull-request/builds/280358) was flaky 77f20c9413f897deb865c318c20d1e1d1b08649a
* :green_heart: [Build #278234](https://buildkite.com/elastic/kibana-pull-request/builds/278234) succeeded c3c8b8900df4bbb6a060da5aab3997f6506e7892

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283225","number":283225,"commit":"c37488b6c6f446f2fabefe921200e20e2b76e120"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280358","number":280358,"commit":"77f20c9413f897deb865c318c20d1e1d1b08649a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/278234","number":278234,"commit":"c3c8b8900df4bbb6a060da5aab3997f6506e7892"}],"number":283225}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

ymao1 committed:
Merging in main

ymao1 committed:
Merge branch 'alerting/alerts-deletion' of github.com:elastic/kibana into alerting/alerts-deletion

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283450)
* Commit: 445646d31d5f6b365ca80e1384737dc5d2f7ad60
* [Cloud Deployment](https://kibana-pr-211384.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #283225](https://buildkite.com/elastic/kibana-pull-request/builds/283225) failed c37488b6c6f446f2fabefe921200e20e2b76e120
* :yellow_heart: [Build #280358](https://buildkite.com/elastic/kibana-pull-request/builds/280358) was flaky 77f20c9413f897deb865c318c20d1e1d1b08649a
* :green_heart: [Build #278234](https://buildkite.com/elastic/kibana-pull-request/builds/278234) succeeded c3c8b8900df4bbb6a060da5aab3997f6506e7892

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283450","number":283450,"commit":"445646d31d5f6b365ca80e1384737dc5d2f7ad60"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283225","number":283225,"commit":"c37488b6c6f446f2fabefe921200e20e2b76e120"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280358","number":280358,"commit":"77f20c9413f897deb865c318c20d1e1d1b08649a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/278234","number":278234,"commit":"c3c8b8900df4bbb6a060da5aab3997f6506e7892"}],"number":283450}
buildkite-pr-comment-->

jcger committed:
[Response Ops][Alerting] Alert Deletion - Persist settings (#211488)

## Summary

https://github.com/elastic/kibana/issues/209258 updates the settings
endpoint to also be able to get/set the alert deletion settings. The
alert deletion setting should make use of this new endpoint to load its
initial data and store any user update.

> [!WARNING]
> This will be merged into a feature branch.


## QA:
Activate the feature flag
```
# config/kibana.dev.yml
xpack.trigger_actions_ui.enableExperimental: ['alertDeletionSettingsEnabled']
```
Follow these steps:
- Go to rules
- Click on settings
- Change the alert deletion settings
- Click on save
- Reload and check the settings kept the values

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into alerting/alerts-deletion

ymao1 committed:
Merge

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285698)
* Commit: f6fbb6332779494223c454a05c87c56828555fb7
* [Cloud Deployment](https://kibana-pr-211384.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #283450](https://buildkite.com/elastic/kibana-pull-request/builds/283450) succeeded 445646d31d5f6b365ca80e1384737dc5d2f7ad60
* :broken_heart: [Build #283225](https://buildkite.com/elastic/kibana-pull-request/builds/283225) failed c37488b6c6f446f2fabefe921200e20e2b76e120
* :yellow_heart: [Build #280358](https://buildkite.com/elastic/kibana-pull-request/builds/280358) was flaky 77f20c9413f897deb865c318c20d1e1d1b08649a
* :green_heart: [Build #278234](https://buildkite.com/elastic/kibana-pull-request/builds/278234) succeeded c3c8b8900df4bbb6a060da5aab3997f6506e7892

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285698","number":285698,"commit":"f6fbb6332779494223c454a05c87c56828555fb7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283450","number":283450,"commit":"445646d31d5f6b365ca80e1384737dc5d2f7ad60"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283225","number":283225,"commit":"c37488b6c6f446f2fabefe921200e20e2b76e120"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/280358","number":280358,"commit":"77f20c9413f897deb865c318c20d1e1d1b08649a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/278234","number":278234,"commit":"c3c8b8900df4bbb6a060da5aab3997f6506e7892"}],"number":285698}
buildkite-pr-comment-->
